src/coreclr/jit/codegen.h

```diff
@@ -329,16 +329,16 @@ class CodeGen final : public CodeGenInterface
         }
     };
 
-    static void genBuildRegPairsStack(regMaskTP regsMask, ArrayStack<RegPair>* regStack);
+    static void genBuildRegPairsStack(regMaskOnlyOne regsMask, ArrayStack<RegPair>* regStack);
     static void genSetUseSaveNextPairs(ArrayStack<RegPair>* regStack);
 
-    static int genGetSlotSizeForRegsInMask(regMaskTP regsMask);
+    static int genGetSlotSizeForRegsInMask(regMaskOnlyOne regsMask);
 
-    void genSaveCalleeSavedRegisterGroup(regMaskTP regsMask, int spDelta, int spOffset);
-    void genRestoreCalleeSavedRegisterGroup(regMaskTP regsMask, int spDelta, int spOffset);
+    void genSaveCalleeSavedRegisterGroup(regMaskOnlyOne regsMask, int spDelta, int spOffset);
+    void genRestoreCalleeSavedRegisterGroup(regMaskOnlyOne regsMask, int spDelta, int spOffset);
 
-    void genSaveCalleeSavedRegistersHelp(regMaskTP regsToSaveMask, int lowestCalleeSavedOffset, int spDelta);
-    void genRestoreCalleeSavedRegistersHelp(regMaskTP regsToRestoreMask, int lowestCalleeSavedOffset, int spDelta);
+    void genSaveCalleeSavedRegistersHelp(AllRegsMask regsToSaveMask, int lowestCalleeSavedOffset, int spDelta);
```

Comment:
Is there any particular reason this needs to be `AllRegsMask` rather than splitting it into 3 calls, one for each mask?

Presumably that's functionally all that is happening internally, is it's functionally doing 2-3 separate loops?

Alternatively, this should be passable by const ref to avoid the shadow copies (Andy pointed this out for many returns, but I think it equally applies to inputs).